### PR TITLE
[Perf] adding a performance test for runtime.Pipeline

### LINKED ISSUE #17213: [Perf] Port pipeline client GET test
* Python example: https://github.com/Azure/azure-sdk-for-python/blob/bedff8a444413458c84bcc8d08caa7b7e84f2b80/tools/azure-devtools/src/azure_devtools/perfstress_tests/system_perfstress/pipeline_client_get_test.py

### PR DESCRIPTION
**Pull request by seankane-msft**
Currently blocked by #17231

Closes #17213

### CODE REVIEW

File: sdk/azcore/testdata/perf/client_get.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+type clientGETTestOptions struct {
+	url string
+}
+
+var clientGetOpts clientGETTestOptions = clientGETTestOptions{url: ""}
+
+// sleepTestRegister is called once per process
+func clientTestRegister() {
+	flag.StringVar(&clientGetOpts.url, "url", "", "URL to send a GET request")
```
Comment (richardpark-msft) [inline sdk/azcore/testdata/perf/client_get.go:26]: What do you typically use here now? (maybe the bing.com default was never actually triggered in practice/)

**Review by richardpark-msft — APPROVED** [review]
Looks okay and pretty minor.

File: sdk/azcore/testdata/perf/client_get.go (hunk 1 above)
Comment (seankane-msft) [reply sdk/azcore/testdata/perf/client_get.go:26]: When I'm testing for validation I have used bing.com. I'm not sure what @mikeharder uses in the real performance tests.

Comment (mikeharder) [reply sdk/azcore/testdata/perf/client_get.go:26]: We currently do not run an HttpClient GET test in the perf automation.  If I were to run an HttpClient GET test manually, I would use a super-fast "hello world" web server (like https://github.com/mikeharder/MinKestrel/tree/master/MinKestrel) running on a VM in the same virtual network as the client VM.

**Pull request merged**